File: wrench-app/src/main/java/com/izettle/wrench/database/WrenchDatabase.java
package com.izettle.wrench.database;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

@Database(entities = {WrenchApplication.class, WrenchConfiguration.class, WrenchConfigurationValue.class, WrenchPredefinedConfigurationValue.class, WrenchScope.class}, version = 3)
@TypeConverters({RoomDateConverter.class})
public abstract class WrenchDatabase extends RoomDatabase {

    public abstract WrenchApplicationDao applicationDao();

    public abstract WrenchConfigurationDao configurationDao();

    public abstract WrenchConfigurationValueDao configurationValueDao();

    public abstract WrenchPredefinedConfigurationValueDao predefinedConfigurationValueDao();

    public abstract WrenchScopeDao scopeDao();


}

File: wrench-app/src/main/java/com/izettle/wrench/database/WrenchDatabase.kt
package com.izettle.wrench.database

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters

@Database(entities = [WrenchApplication::class, WrenchConfiguration::class, WrenchConfigurationValue::class, WrenchPredefinedConfigurationValue::class, WrenchScope::class], version = 3)
@TypeConverters(RoomDateConverter::class)
abstract class WrenchDatabase : RoomDatabase() {

    abstract fun applicationDao(): WrenchApplicationDao

    abstract fun configurationDao(): WrenchConfigurationDao

    abstract fun configurationValueDao(): WrenchConfigurationValueDao

    abstract fun predefinedConfigurationValueDao(): WrenchPredefinedConfigurationValueDao

    abstract fun scopeDao(): WrenchScopeDao


}
